### PR TITLE
Add ability to set loadbalancer and tag instances.

### PR DESCRIPTION
**Pull request by awong-dev**
Also fixes CR/LF problems.

### CODE REVIEW

**Comment by awong-dev** [comment]
@orischwartz This is to support https://github.com/department-of-veterans-affairs/devops/pull/233.  You might want to sync with Ken on what should be moved into common roles, if the loadbalancer config should be just merged into the deploy script, and if the naming conventions make sense.

File: aws-deploy/tasks/main.yml [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
+    launch_config_name: "{{ launch_config_name }}"
+    load_balancers: "{{ app_loadbalancers }}"
+    max_size: "{{ max_size | default(2) }}"
+    min_size: "{{ min_size | default(2) }}"
+    name: "{{ auto_scaling_group_name }}"
+    region: "{{ region }}"
+    replace_all_instances: yes
+    replace_batch_size: "{{ replace_batch_size | default(1) }}"
+    state: present
+    tags: "{{ app_tags | default(null) }}"
```
Comment (orischwartz) [inline aws-deploy/tasks/main.yml]: @awong-dev the `default(null)` syntax actually doesn't work. Any other ideas? I dont think we can both use this role as intended. 

* We update an existing ASG. Any params that are omitted default to their existing value.
* Looks like you want to create the ASG from scratch and specify all params each time.

I don't know of a good way to reconcile these other than to maybe create two tasks:

```yml
- name: update existing asg
  ...
  when: create_asg is not defined

- name: create new asg
  when: create_asg is defined
```

Comment (awong-dev) [inline aws-deploy/tasks/main.yml]: Just found this:

http://stackoverflow.com/questions/33394333/how-to-add-a-key-value-to-a-dict-in-an-ansible-playbook-conditional

Ick...seems like this isn't something easily done.

IMO, we should have only one and we should *always* specify all the options. The mantra in ansible is to make the playbooks idempotent so setup and update should look the same. That's why it's declarative "roles" instead of imperative actions.

If you look at the convention I've been toying with here:
  https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/roles/vetsgov/aws-config/defaults/main.yml

Basically there is one role that has *all* of the site configuration as a static key/value pair. This role is then loaded via meta/main.yml first in pretty much every other role which lets us just reference stuff via `aws_config[env].subnet_ids`.

In that work, rather than aws-deploy, you'd name the role something like aws-frontend-stack, and the goal is to always ensure the whole lb/asg/lc/ami/sg setup was in sync with whatever was specified in aws_config.

Deploy then becomes, update aws_config. Commit. Run play. This creates a nice audit log....

Thoughts?

Comment (awong-dev) [inline aws-deploy/tasks/main.yml]: Aaand...I found a workaround:

`{{ optional_value | default(None) or omit }}` does what I was trying to do with `default(null)`

Comment (orischwartz) [inline aws-deploy/tasks/main.yml]: :+1: Awesome. And yes, I do agree with you that ideally everything should be in scripts in declarative format but there are certain infrastructure components I'd prefer to avoid touching on every deploy, and ELB is one of them.

I was going to take a stab at adopting your approach but this will help in the interim.

Comment (awong-dev) [inline aws-deploy/tasks/main.yml]: Yay!

I'm playing with the elb setup ansible scripts right now and ansible seems smart enough to look at the configuration and only modify what's different...I think having it declarative is actually safe in this situation...
